keanu-python/keanu/stats.py

```diff
@@ -0,0 +1,21 @@
+from typing import List
+
+from numpy import ndarray, fromiter, issubdtype, hstack, floating
+from py4j.java_gateway import java_import
+
+from keanu.vartypes import numpy_types
+from .context import KeanuContext
+
+k = KeanuContext()
+
+java_import(k.jvm_view(), "io.improbable.keanu.algorithms.SampleStats")
+
+
+def autocorrelation(arg: List[numpy_types]) -> ndarray:
+    if not issubdtype(arg[0].dtype, floating):
```

Comment:
Good point, we should check all of them in case users use this in a way that doesn't get the samples from our sampling. I will change this and the shape check.